test/browserTests/street-geo-test-bdd.mjs

```diff
@@ -0,0 +1,96 @@
+describe('street-geo component', function() {
+  let el;
+
+  before((done) => {
+    const scene = document.createElement('a-scene');
+    document.body.appendChild(scene);
+    el = document.createElement('a-entity');
+    el.setAttribute('id', 'street-geo-test');
+    el.setAttribute('street-geo', {
+        longitude: 10,
+        latitude: 20,
+        elevation: 30,
+        maps: 'mapbox2d'
+    });
+    scene.appendChild(el);    
+
+    setTimeout(() => {
+      done();
+    }, 500);
+  });
+
+  it('should create a mapbox2d element', () => {
+    const mapbox2dElement = el.querySelector('[data-layer-name="Mapbox Satellite Streets"]');
+    expect(mapbox2dElement).to.exist;
+  });
+
+  it('should create a google3d element and delete mapbox2d element', (done) => {
+    const mapbox2dElement = el.querySelector('[data-layer-name="Mapbox Satellite Streets"]');
+    const google3dElement = el.querySelector('[data-layer-name="Google 3D Tiles"]');
+    
+    expect(mapbox2dElement).to.exist;
+
+    el.setAttribute('street-geo', 'maps', 'google3d');
+
+    setTimeout(() => {
+      const updatedMapbox2dElement = el.querySelector('[data-layer-name="Mapbox Satellite Streets"]');
+      const updatedGoogle3dElement = el.querySelector('[data-layer-name="Google 3D Tiles"]');
+      
+      expect(updatedMapbox2dElement).to.not.exist;
+      expect(updatedGoogle3dElement).to.exist;
+
+      done();
+    }, 1000);
+  });
+
+  it('should create both mapbox2d and google3d elements', (done) => {
+    el.setAttribute('street-geo', 'maps', 'mapbox2d, google3d');
+
+    setTimeout(() => {
+      const mapbox2dElement = el.querySelector('[data-layer-name="Mapbox Satellite Streets"]');
+      const google3dElement = el.querySelector('[data-layer-name="Google 3D Tiles"]');
+      expect(mapbox2dElement).to.exist;
+      expect(google3dElement).to.exist;
+      done();
+    }, 1000);
```

Comment:
Do you really need to wait 1s? We don't have much tests so that may be fine but if we can avoid that it would be better. You probably just need two setTimeout to be sure the street-geo is initialized and the child entity with components are created and initialized.
```js
setTimeout(() => {
   setTimeout(() => {
     const mapbox2dElement = el.querySelector('[data-layer-name="Mapbox Satellite Streets"]');
      const google3dElement = el.querySelector('[data-layer-name="Google 3D Tiles"]');
      expect(mapbox2dElement).to.exist;
      expect(google3dElement).to.exist;
      done();
  });
});
```

Reply:
Thanks for the tip!